purchases/src/main/kotlin/com/revenuecat/purchases/CustomerInfoHelper.kt

```diff
@@ -121,14 +125,29 @@ internal class CustomerInfoHelper(
             appInBackground,
             { info ->
                 log(LogIntent.RC_SUCCESS, CustomerInfoStrings.CUSTOMERINFO_UPDATED_FROM_NETWORK)
+                offlineEntitlementsManager.resetOfflineCustomerInfoCache()
                 cacheCustomerInfo(info)
                 sendUpdatedCustomerInfoToDelegateIfChanged(info)
                 dispatch { callback?.onReceived(info) }
             },
-            { error, _ ->
+            { error, isServerError ->
                 Log.e("Purchases", "Error fetching customer data: $error")
                 deviceCache.clearCustomerInfoCacheTimestamp(appUserID)
-                dispatch { callback?.onError(error) }
+                if (shouldCalculateOfflineCustomerInfo(isServerError, appUserID)) {
+                    offlineEntitlementsManager.calculateAndCacheOfflineCustomerInfo(
+                        appUserID,
+                        onSuccess = { offlineComputedCustomerInfo ->
+                            // TODO Improve logs
+                            warnLog("Using offline computed customer info.")
+                            dispatch { callback?.onReceived(offlineComputedCustomerInfo) }
+                        },
+                        onError = {
+                            dispatch { callback?.onError(error) }
```

Comment:
Note how on errors with calculating the offline entitlements, I propagate the original backend error, and not the error calculating the offline entitlements. I think that's preferable. Lmk if you think otherwise.

Reply:
agreed, if we try but fail to compute offline we should just send whatever the backend sent, no value added in doing anything else

Reply:
Maybe renaming `error` to `backendError` would make this more clear for the future